cmake/macros/macro_deal_ii_query_git_information.cmake

```diff
@@ -0,0 +1,113 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2015 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# This file implements the DEAL_II_QUERY_GIT_INFORMATION macro, which is
+# part of the deal.II library.
+#
+# Usage:
+#       DEAL_II_QUERY_GIT_INFORMATION()
+#
+# This will try to gather information about current branch, as well as
+# short and long revision. If ${CMAKE_SOURCE_DIR} is the root of a git
+# repository the following variables will be populated:
+#
+#       GIT_BRANCH
+#       GIT_REVISION
+#       GIT_SHORTREV
+#
+# If this macro is called within the deal.II build system the variables are
+# prefixed with DEAL_II_:
+#
+#       DEAL_II_GIT_BRANCH
+#       DEAL_II_GIT_REVISION
+#       DEAL_II_GIT_SHORTREV
+#
+
+MACRO(DEAL_II_QUERY_GIT_INFORMATION)
+
+  MESSAGE(STATUS "Query git repository information.")
+
+  #
+  # If DEAL_II_BASE_NAME is defined and DEAL_II_PROJECT_CONFIG_INCLUDED was
+  # not set, we assume that we are called from within the deal.II build
+  # system. In this case we prepend all variables by "DEAL_II_"
+  #
+  IF( DEFINED DEAL_II_BASE_NAME AND
+      NOT DEFINED DEAL_II_PROJECT_CONFIG_INCLUDED )
+    SET(_prefix "DEAL_II_")
+  ELSE()
+    SET(_prefix "")
+  ENDIF()
```

Comment:
> What's the rationale to omit the prefix in user-land? I would assume
> that there it's even more important to have the prefix because they
> may be querying the git hash of multiple libraries they depend on, so
> the prefix would help distinguish which package's version they are
> dealing with.

This macro always queries git information from the current
CMAKE_SOURCE_DIR. Consequently, in user projects it queries for branch
and revision from the user project, not from deal.II. Thus, no prefix.


This reminds me of the fact that I totally forgot to export the git
information in deal.IIConfig.cmake and Make.global_options. One second.